Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardEditViewModel.swift

```diff
@@ -7,35 +7,109 @@ import SwiftUI
 import Common
 import Storage
 
+enum CreditCardEditState: String, Equatable, CaseIterable {
+    case add
+    case edit
+    case view
+
+    enum leftBarButtonState: String, Equatable, CaseIterable {
+        case Close
+        case Cancel
+
+        var title: String {
+            switch self {
+            case .Cancel:
+                return String.CreditCard.EditCard.CancelNavBarButtonLabel
+            case .Close:
+                return String.CreditCard.EditCard.CloseNavBarButtonLabel
+            }
+        }
+    }
+
+    enum rightBarButtonState: String, Equatable, CaseIterable {
+        case Save
+        case Edit
+
+        var title: String {
+            switch self {
+            case .Save:
+                return String.CreditCard.EditCard.SaveNavBarButtonLabel
+            case .Edit:
+                return String.CreditCard.EditCard.EditNavBarButtonLabel
+            }
+        }
+    }
+
+    var title: String {
+        switch self {
+        case .add:
+            return String.CreditCard.EditCard.AddCreditCardTitle
+        case .view:
+            return String.CreditCard.EditCard.ViewCreditCardTitle
+        case .edit:
+            return String.CreditCard.EditCard.EditCreditCardTitle
+        }
+    }
+
+    var leftBarBtn: leftBarButtonState {
+        switch self {
+        case .add, .view:
+            return .Close
+        case .edit:
+            return .Cancel
+        }
+    }
+
+    var rightBarBtn: rightBarButtonState {
```

Comment:
I saw your comment about Unit test in the other PR I just wanted to mention that with the state and the buttons there is new cases to test